clams/source/__init__.py

```diff
@@ -200,12 +201,14 @@ def generate_source_mmif(documents, prefix=None, **ignored):
             raise ValueError(
                 f'Invalid MIME types, or no MIME type and/or path provided, in argument {doc_id-1} to source'
             )
-        if prefix and path.isabs(location):
-            raise ValueError(f"when prefix is used, file location must not be an absolute path; given \"{location}\".")
-        elif not prefix and not path.isabs(location):
-            raise ValueError(f'file location must be an absolute path, or --prefix must be used; given \"{location}\".')
+        if (prefix or scheme) and path.isabs(location):
+            raise ValueError(f"when prefix or scheme is used, file location must not be an absolute path; given \"{location}\".")
+        elif not (prefix or scheme) and not path.isabs(location):
+            raise ValueError(f'file location must be an absolute path, or --prefix or --scheme must be used; given \"{location}\".')
         elif prefix and not path.isabs(location):
             location = path.join(prefix, location)
+        elif scheme and not path.isabs(location):
+            location = path.join(scheme, location)
```

Comment:
`os.path.join` uses `\` on windows and `/` on posix as the join delimiter, both of which are not what we want here. We should use literally `://` to join scheme and location.